gnucash/report/report.scm

```diff
@@ -37,6 +37,21 @@
 (gnc:module-load "gnucash/html" 0)
 (gnc:module-load "gnucash/gnome-utils" 0)
 
+
+(export guard-html-chart)
+(define-syntax guard-html-chart
+  ;; this macro applied to old html-bar/line/scatter/pie apis will
+  ;; guard a report writer from passing html-chart objects. this
+  ;; should be removed in 5.x series.
+  (syntax-rules ()
+    ((_ api)
+     (let ((old-api api))
+       (set! api
+         (lambda args
+           (if (and (pair? args) (gnc:html-chart? (car args)))
+               (gnc:error "trying to use old-api on html-chart object.")
```

Comment:
I think a gnc:warning is more appropriate: gnc:error is meant to inform the user the requested command can't be executed at all and hence the expected result won't be achieved. In this case you have a fallback that still does what the user is expecting, but that fallback will eventually go away. And we want to warn the user well in advance.

I'd also try to write the message in a way it is more conductive to a solution. This only tells something is wrong it doesn't help much in making it right. Remember the user/developer may not be very deep into our api as you and I would be.

For example, mention the specific function that was called (if possible), and indicate it's deprecated. And explain in the new api chart properties are set via xyz or point at some ready to use documentation or examples.